tests/unit/test_dataset_geo.py

```diff
@@ -537,8 +537,9 @@ def update_callback(request, snoop):
                 "record": {"geom": {"point": [1, 1]}, "id1": "2", "id2": "b"},
             },
         ]
-        expected = "\n".join(json.dumps(update) for update in updates)
-        actual = snoop["payload"]
+        expected = [json.dumps(update).replace('"', "'") for update in updates]
+        self.maxDiff = None
+        actual = [a.decode("utf8") for a in list(snoop["payload"])]
```

Comment:
no need to use `list` if you are immediately going to use a list comprehension:

```suggestion
        actual = [a.decode("utf8") for a in snoop["payload"]]
```

This also goes for the other occurrence of `snoop["payload"]`